modules/integration/tests-common/admin-clients/src/main/java/org/wso2/mb/integration/common/clients/operations/utils/AndesClientOutputParser.java

```diff
@@ -243,6 +244,104 @@ public boolean transactedOperations(long operationOccurredIndex) throws FileNotF
     }
 
     /**
+     * Check whether all the messages are received in-order when subscriber rollbacks
+     * after certain number of messages.
+     *
+     * @param messagesPerRollback Number of messages received per rollback
+     * @return Result as true if transacted rollback operation has successfully
+     * preserved order of all messages.
+     */
+    public boolean checkIfTransactedRollbackPreservesOrder(long messagesPerRollback)
+            throws FileNotFoundException {
+
+        boolean result = true;
+        int count = 0;
+        Long messageIdentifier;
+
+        int rollbackBatchIteration = 1;
+        boolean isFirstMessageBatch = true;
+        BufferedReader br = null;
+
+        Map<Integer, Long> firstMessageBatch = new HashMap<Integer, Long>();
+        try {
+            br = new BufferedReader(new FileReader(filePath));
```

Comment:
https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/LineIterator.html would have simplified the logic.